### PR TITLE
daemon, store: switch to new store APIs in snapd

### PR DESCRIPTION
**Pull request by pete-woods**

### CODE REVIEW

**Comment by pete-woods** [comment]
I definitely agree re the debt. I was trying to keep the diff to as readable a level as possible, as there are a lot of little details that could have been easily missed when making these changes. There are a few follow up branches after this, to do things like clean out the /v2/buy/methods endpoint, at very least.

File: store/store.go [hunk 1]
```diff
@@ -773,31 +765,16 @@ func (s *Store) decoratePurchases(snaps []*snap.Info, channel string, user *auth
...
+	purchases, err := s.getPurchasesFromURL(s.purchasesURI, channel, user)
 	if err != nil {
 		return err
 	}
 
 	// Group purchases by snap ID.
-	purchasesByID := make(map[string][]*purchase)
-	for _, purchase := range purchases {
-		purchasesByID[purchase.SnapID] = append(purchasesByID[purchase.SnapID], purchase)
+	purchasesByID := make(map[string]bool)
```
Comment (pete-woods) [inline store/store.go]: John: After discussion, we resolved this misunderstanding, and agreed on naming for the map.

File: store/store.go [hunk 2]
```diff
@@ -664,66 +664,58 @@ func (s *Store) extractSuggestedCurrency(resp *http.Response) {
 	}
 }
 
-// purchase encapsulates the purchase data sent to us from the software center agent.
+// order encapsulates the purchase data sent to us from the software center agent.
```
Comment (pete-woods) [inline store/store.go]: Done

File: store/store.go [hunk 3]
```diff
@@ -1581,7 +1525,7 @@ type storeCustomer struct {
 // ReadyToBuy returns nil if the user's account has accepted T&Cs and has a payment method registered, and an error otherwise
 func (s *Store) ReadyToBuy(user *auth.UserState) error {
 	if user == nil {
-		return ErrInvalidCredentials
```
Comment (chipaca) [inline store/store.go]: good catch

File: daemon/api.go [hunk 4]
```diff
@@ -1782,6 +1782,44 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
...
 }
 
+func convertBuyError(err error) Response {
+	switch err {
+	default:
+		return InternalError("%v", err)
+	case store.ErrInvalidCredentials:
+		return Unauthorized(err.Error())
+	case store.ErrUnauthenticated:
+		return SyncResponse(&resp{
```
Comment (chipaca) [inline daemon/api.go]: (and yes, I know there already are a bunch of these, so if you want to leave it I'll do it in a branch all of its own)

File: daemon/api.go [hunk 5]
```diff
@@ -1782,6 +1782,44 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
...
+	case store.ErrNoPaymentMethods:
+		return SyncResponse(&resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Message: err.Error(),
+				Kind:    errorKindNoPaymentMethods,
+			},
+			Status: http.StatusBadRequest,
+		}, nil)
+	case nil:
```
Comment (chipaca) [inline daemon/api.go]: followup nit: in the spirit of "exit early", this is usually the first case in the switch

File: daemon/api.go [hunk 6]
```diff
@@ -1829,32 +1872,9 @@ func readyToBuy(c *Command, r *http.Request, user *auth.UserState) Response {
...
-			Type: ResponseTypeError,
-			Result: &errorResult{
-				Message: err.Error(),
-				Kind:    errorKindNoPaymentMethods,
-			},
-			Status: http.StatusBadRequest,
-		}, nil)
-	case nil:
-		// continue
+	resp := convertBuyError(err)
```
Comment (chipaca) [inline daemon/api.go]: ditto here wrt moving it into the if

File: daemon/api.go (hunk 4 above)
Comment (pete-woods) [inline daemon/api.go]: agreed to fix this in a followup

**Pull request merged**